hcl2template/utils.go

```diff
@@ -124,3 +124,48 @@ func PrintableCtyValue(v cty.Value) string {
 	str := repl.FormatResult(gval)
 	return str
 }
+
+func ConvertPluginConfigValueToHCLValue(v interface{}) (cty.Value, error) {
+	var buildValue cty.Value
+	switch v := v.(type) {
+	case bool:
+		buildValue = cty.BoolVal(v)
+	case string:
+		buildValue = cty.StringVal(v)
+	case uint8:
+		buildValue = cty.NumberUIntVal(uint64(v))
+	case float64:
+		buildValue = cty.NumberFloatVal(v)
+	case int64:
+		buildValue = cty.NumberIntVal(v)
+	case uint64:
+		buildValue = cty.NumberUIntVal(v)
+	case []string:
+		vals := make([]cty.Value, len(v))
+		for i, ev := range v {
+			vals[i] = cty.StringVal(ev)
+		}
+		buildValue = cty.ListVal(vals)
+	case []uint8:
+		vals := make([]cty.Value, len(v))
+		for i, ev := range v {
+			vals[i] = cty.NumberUIntVal(uint64(ev))
+		}
+		buildValue = cty.ListVal(vals)
+	case []int64:
+		vals := make([]cty.Value, len(v))
+		for i, ev := range v {
+			vals[i] = cty.NumberIntVal(ev)
+		}
+		buildValue = cty.ListVal(vals)
+	case []uint64:
+		vals := make([]cty.Value, len(v))
+		for i, ev := range v {
+			vals[i] = cty.NumberUIntVal(ev)
+		}
+		buildValue = cty.ListVal(vals)
```

Comment:
I'm not sure whether a build var can be an empty list or not, but is not possible to create a cty.ListVal from an empty list, so this will fail when `vals` is empty.

```suggestion
                if len(vals) == 0 {
                    return cty.ListValEmpty(cty.Number)
                }
		buildValue = cty.ListVal(vals)
```

Reply:
I know the other function doesn't deal with this case, but I wanted to leave it as it was to avoid changing code that I wasn't using with data sources

Reply:
Oh woa, I didn't know:

https://github.com/hashicorp/packer/blob/77c3c12244f3847762788a30feb3f6ec9f1617ff/vendor/github.com/zclconf/go-cty/cty/value_init.go#L155-L158

Reply:
Oh right I think that's because a ListVal needs a type, `ListValEmpty` exists too, but you have to give it a type

Reply:
Yes! Only tuple and object exist empty without a type, but all the others require a type, even Map if I'm not wrong.

Reply:
great catch, thanks Moss.